### PR TITLE
Add 'Give feedback' button and settings panel

### PR DESCRIPTION
**Pull request by lesleyjanenorton**
Here we:
- Add a 'Give feedback' button to the bottom of the main panel
- Add a settings panel that is opened by clicking the settings icon 
  - Some outbound URLs are incorrect because we don't have final URLs
  - UTM params to come later

### CODE REVIEW

File: src/ui/browserAction/popupPage.js [hunk 1]
```diff
@@ -350,6 +367,50 @@ export class BrowserActionPopup extends LitElement {
       slot="left"
     ></mz-iconlink>`;
   }
+
+  static createSettingsPanel(onResult = () => {}) {
+    const viewElement = document.createElement("section");
+    viewElement.classList = ["settings-panel"];
+    viewElement.dataset.title = "Settings";
+    const linkList = document.createElement("ul");
```
Comment (strseb) [inline src/ui/browserAction/popupPage.js:375]: I'm not going to enforce lit for all dom' things. but still consider: 

```js
let linklist = [
 {linkTitle, callback, iconId}
]
 const viewElement = document.createElement("section");
    viewElement.classList = ["settings-panel"];
    viewElement.dataset.title = "Settings";


render(html`
<ul>
 ${linklist.map(link => html`
   <li>
    <button class="${link.iconId}" @onClick=${link.callback}> ${link.Title}</button>
   </li>
  `)}
</ul>
`, viewElement);



```

File: src/ui/browserAction/popupPage.js [hunk 2]
```diff
@@ -350,6 +367,50 @@ export class BrowserActionPopup extends LitElement {
       slot="left"
     ></mz-iconlink>`;
   }
+
+  static createSettingsPanel(onResult = () => {}) {
```
Comment (strseb) [inline src/ui/browserAction/popupPage.js:371]: onResult is unused?

Comment (lesleyjanenorton) [reply src/ui/browserAction/popupPage.js:381]: idk how I'm just seeing this - thanks!

File: src/ui/browserAction/popupPage.js (hunk 1 above)
Comment (lesleyjanenorton) [reply src/ui/browserAction/popupPage.js]: done!

**Review by strseb — APPROVED** [review]
r+wc,

File: src/components/iconbutton.js [hunk 3]
```diff
@@ -26,14 +26,21 @@ export class IconLinkButton extends LitElement {
...
   };
   constructor() {
     super();
     this.href = "";
     this.alt = "";
     this.icon = "";
+    this.eventLabel = "";
   }
   onClick() {
+    if (this.eventLabel) {
```
[the remark this answers is not in the document]
Comment (strseb) [reply src/components/iconbutton.js:39]: Or we remove the default click handler at all and we replace that usecase with a factory generating those buttons: 
```js 
functionLinkBTN(href, icon){
 const handler ()=>{
 browser.tabs.create({
               url: href,
             });
    window.close();

 }
 return html`
   <mz-iconbutton @click=${handler}  icon ></>

 `
}

```

[the remark this answers is not in the document]
Comment (lesleyjanenorton) [reply src/components/iconbutton.js]: done!

File: src/components/styles.js [hunk 4]
```diff
@@ -151,17 +152,20 @@ export const ghostButtonStyles = css`
...
+  #settingsList button:focus-visible {
     border: 2px solid lch(from var(--button-ghost-bg-color) l c h / 0.5);
   }
 
-  .ghost-btn:active::before {
+  .ghost-btn:active::before,
+  #settingsList button:active {
     background: lch(from var(--button-ghost-bg-color) l c h / 0.15);
-    border: none;
+    // border: none;
```
[the remark this answers is not in the document]
Comment (lesleyjanenorton) [reply src/components/styles.js]: gone!

**Pull request merged**